pkg/provider/aws/client.go

```diff
@@ -78,6 +79,9 @@ func (c *Client) GetAddrs(ctx context.Context) ([]string, error) {
 		if err != nil {
 			return nil, err
 		}
+		if strings.TrimSpace(addr) == "" {
```

Comment:
The TrimSpace here is not necessary, however, ensuring that the address received is a valid ipv4 address would be a good thing to check for.  The check would attempt to parse ipv4 and the second check, should it be a valid ipv4 address, will be to check to ensure it is also a non-loopback address.